pkg/logs/input/eventlog/launcher.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package eventlog
+
+import (
+	log "github.com/cihub/seelog"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/auditor"
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/pipeline"
+	"github.com/DataDog/datadog-agent/pkg/logs/restart"
+)
+
+// Launcher is in charge of starting and stopping windows event logs tailers
+type Launcher struct {
+	sources          []*config.LogSource
+	pipelineProvider pipeline.Provider
+	auditor          *auditor.Auditor
+	tailers          map[string]*Tailer
+}
+
+// New returns a new Launcher.
+func New(sources []*config.LogSource, pipelineProvider pipeline.Provider, auditor *auditor.Auditor) *Launcher {
+	windowsEventSources := []*config.LogSource{}
+	for _, source := range sources {
+		if source.Config.Type == config.EventLogType {
+			windowsEventSources = append(windowsEventSources, source)
+		}
+	}
+	return &Launcher{
+		sources:          windowsEventSources,
+		pipelineProvider: pipelineProvider,
+		auditor:          auditor,
+		tailers:          make(map[string]*Tailer),
+	}
+}
+
+// Start starts new tailers.
+func (l *Launcher) Start() {
+	log.Info("Start tailing eventlog")
+	availableChannels, err := EnumerateChannels()
+	if err != nil {
+		log.Debug("Could not list eventlog channels: ", err)
+	} else {
+		log.Debug("Found available eventlog channels: ", availableChannels)
+	}
+
+	for _, source := range l.sources {
+		identifier := Identifier(source.Config.ChannelPath, source.Config.Query)
+		if _, exists := l.tailers[identifier]; exists {
+			// tailer already setup
+			continue
+		}
+		tailer, err := l.setupTailer(source)
+		if err != nil {
+			log.Info("Could not set up eventlog tailer: ", err)
+		} else {
+			l.tailers[identifier] = tailer
+		}
+	}
+}
+
+// Stop stops all active tailers
+func (l *Launcher) Stop() {
+	stopper := restart.NewParallelStopper()
+	for _, tailer := range l.tailers {
+		stopper.Add(tailer)
+		delete(l.tailers, tailer.Identifier())
+	}
+	stopper.Stop()
+}
+
+// sanitizedConfig sets default values for the config
+func (l *Launcher) sanitizedConfig(sourceConfig *config.LogsConfig) *Config {
```

Comment:
I think you could also add a default value with `viper`.

Reply:
right, originally we wanted to override the default config in some cases, but we ended up not doing it, so a default value works now

Reply:
Actually, today we don't separate configs across integrations, so if I set it as default for one config it'd be set for all configs. I think I'd rather keep things separated, it doesn't cost much and makes it less error prone